modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/trait/DistributionFunction.java

```diff
@@ -135,6 +156,12 @@ public static boolean satisfy(DistributionFunction f0, DistributionFunction f1)
         @Override public RelDistribution.Type type() {
             return RelDistribution.Type.ANY;
         }
+
+        /** {@inheritDoc} */
+        @Override public <Row> Destination<Row> destination(ExecutionContext<Row> ctx, AffinityService affinityService,
+            ColocationGroup m, ImmutableIntList k) {
+            throw new AssertionError();
```

Comment:
```suggestion
            throw new UnsupportedOperationException();
```

Reply:
Changed this to `IllegalStateException` since ANY distribution is not supposed to be a real distribution hence the `distribution` method should never be invoked